pkg/handler/cli/article_handler.go

```diff
@@ -0,0 +1,207 @@
+package cli
+
+import (
+	"errors"
+	"fmt"
+	"github.com/Masterminds/sprig"
+	"log"
+	"news-aggregator/pkg/model"
+	"news-aggregator/pkg/parser"
+	"os"
+	"path/filepath"
+	"sort"
+	"strings"
+	"text/template"
+)
+
+// loadData loads articles from the specified files and saves them to the Service.
+func (h *Handler) loadData() error {
+	execDir, err := os.Getwd()
+	if err != nil {
+		log.Fatalf("Error getting current working directory: %v", err)
+	}
+	log.Printf("Current working directory: %s\n", execDir)
+
+	var articles []model.Article
+	files := []string{
+		filepath.Join(execDir, "../../data/abcnews-international-category-19-05-24.xml"),
+		filepath.Join(execDir, "../../data/bbc-world-category-19-05-24.xml"),
+		filepath.Join(execDir, "../../data/washingtontimes-world-category-19-05-24.xml"),
+		filepath.Join(execDir, "../../data/nbc-news.json"),
+		filepath.Join(execDir, "../../data/usatoday-world-news.html"),
+	}
+	articles, err = parser.ParseArticlesFromFiles(files)
+	if err != nil {
+		return errors.New("error parsing articles from files")
+	}
+
+	h.Service.SaveAll(articles)
+	return nil
+}
+
+// filterArticles filters the provided articles based on the provided sources, keywords, and date range.
+// It returns the filtered articles.
+func (h *Handler) filterArticles(sources, keywords, dateStart, dateEnd string) []model.Article {
```

Comment:
if we keep adding more and more filters, this func will be so huge. Pls create an abstraction "filters" and work with it. 
this way it will be extendable
i was explaining about this some time ago